### PR TITLE
Add windows support

### PR DESCRIPTION
**Pull request by morsdyce**
Please test thoroughly on OSX

### CODE REVIEW

File: lib/commands/upgrade.js [hunk 1]
```diff
@@ -1,13 +1,15 @@
 'use strict';
 
+const path          = require('path');
+const Logger        = require('../logger');
```
Comment (ilyagelman) [inline lib/commands/upgrade.js]: You can require relative to root directory
`require('lib/logger');'

Same for the rest of the files

File: lib/commands/upgrade.js [hunk 2]
```diff
@@ -49,8 +51,8 @@ module.exports = function () {
   function upgradeKick() {
     Logger.log('Updating kick...'.white);
 
-    child_process.execSync('npm install -g kick');
```
Comment (ilyagelman) [inline lib/commands/upgrade.js]: Do we still need to require child_process then?

File: lib/utils.js [hunk 3]
```diff
@@ -1,20 +1,26 @@
 'use strict';
 
+const which         = require('which');
 const fs            = require('fs-extra');
 const changeCase    = require('change-case');
 const child_process = require('child_process');
-const path          = require('path');
```
Comment (ilyagelman) [inline lib/utils.js]: Better leave `nodePath` as `path` and rename arguments in functions

File: lib/utils.js [hunk 4]
```diff
@@ -65,6 +71,7 @@ module.exports = class Utils {
   }
 
   static createFile(filename, template) {
+    filename = Utils.getPath(filename);
```
Comment (ilyagelman) [inline lib/utils.js]: A blank line would look good here

File: lib/utils.js [hunk 5]
```diff
@@ -157,23 +174,54 @@ module.exports = class Utils {
   }
 
   static spawnProcess(command, args, environment, options) {
+    if (command.indexOf('/') !== -1) {
```
Comment (ilyagelman) [inline lib/utils.js]: What does it mean? Maybe add a comment

File: lib/utils.js [hunk 6]
```diff
@@ -157,23 +174,54 @@ module.exports = class Utils {
...
     }
 
-    const spawn = require('child_process').spawn;
     const env = Object.create(process.env);
+    if (environment) {
+      env.NODE_ENV = environment;
+    }
 
-    env.NODE_ENV = environment;
+    if ((options || {}).sync) {
```
Comment (ilyagelman) [inline lib/utils.js]: Might be a good idea for a separate function

File: lib/utils.js [hunk 7]
```diff
@@ -157,23 +174,54 @@ module.exports = class Utils {
...
+
+      return spawn(command, args, {
+        stdio: (options || {}).inherit ? 'inherit' : ['pipe', 'pipe', process.stderr],
+        env
+      });
+    }
+  }
+
+  static terminateProcess(childProcess) {
+    var isWin = /^win/.test(process.platform);
```
Comment (ilyagelman) [inline lib/utils.js]: Could use const here
Also please add spacing (blank lines)

File: lib/utils.js [hunk 8]
```diff
@@ -157,23 +174,54 @@ module.exports = class Utils {
...
+    if(!isWin) {
+      childProcess.kill('SIGHUP');
+    } else {
+      const child_process = require('child_process');
+      child_process.exec(`taskkill /PID ${childProcess.pid} /T /F`, function (error, stdout, stderr) {
+        // console.log('stdout: ' + stdout);
```
Comment (ilyagelman) [inline lib/utils.js]: Commented out code

File: lib/commands/upgrade.js (hunk 2 above)
Comment (morsdyce) [inline lib/commands/upgrade.js]: it was later changed to Utils.spawnProcess('npm', ['install', '-g', 'kick'], null, { sync: true });

Comment (ilyagelman) [inline lib/commands/upgrade.js]: Yeah, but you still require it on line 6

**Pull request merged**